File: java/1.7/factory/src/com/pizza/pizzaIngredientFactory/ChicagoPizzaIngredientFactory.java
package com.pizza.pizzaIngredientFactory;

import com.pizza.cheese.Cheese;
import com.pizza.cheese.MozzarellaCheese;
import com.pizza.clams.Clams;
import com.pizza.clams.FrozenClams;
import com.pizza.dough.Dough;
import com.pizza.dough.ThickCrustDough;
import com.pizza.pepperoni.Pepperoni;
import com.pizza.pepperoni.SlicedPepperoni;
import com.pizza.sauce.PlumTomatoSauce;
import com.pizza.sauce.Sauce;
import com.pizza.veggies.BlackOlives;
import com.pizza.veggies.Eggplant;
import com.pizza.veggies.Spinach;
import com.pizza.veggies.Veggies;

public class ChicagoPizzaIngredientFactory 
	implements PizzaIngredientFactory 
{

	public Dough createDough() {
		return new ThickCrustDough();
	}

	public Sauce createSauce() {
		return new PlumTomatoSauce();
	}

	public Cheese createCheese() {
		return new MozzarellaCheese();
	}

	public Veggies[] createVeggies() {
		Veggies veggies[] = { new BlackOlives(), 
		                      new Spinach(), 
		                      new Eggplant() };
		return veggies;
	}

	public Pepperoni createPepperoni() {
		return new SlicedPepperoni();
	}

	public Clams createClam() {
		return new FrozenClams();
	}
}

File: php/5.3/factory/pizzaIngredientFactory/ChicagoPizzaIngredientFactory.php
<?
include_once("./pizzaIngredientFactory/PizzaIngredientFactory.php");

include_once("./dough/ThickCrustDough.php");
include_once("./sauce/PlumTomatoSauce.php");
include_once("./cheese/MozzarellaCheese.php");
include_once("./veggies/BlackOlives.php");
include_once("./veggies/Spinach.php");
include_once("./veggies/Eggplant.php");
include_once("./pepperoni/SlicedPepperoni.php");
include_once("./clams/FrozenClams.php");

class ChicagoPizzaIngredientFactory implements PizzaIngredientFactory {
    public function createDough() {
        return new ThickCrustDough();
    }

    public function createSauce() {
        return new PlumTomatoSauce();
    }

    public function createCheese() {
        return new MozzarellaCheese();
    }

    public function createVeggies() {
        $this->veggies[] = new BlackOlives();
        $this->veggies[] = new Spinach();
        $this->veggies[] = new Eggplant();
        return $this->veggies;
    }

    public function createPepperoni() {
        return new SlicedPepperoni();
    }

    public function createClam() {
        return new FrozenClams();
    }
}
?>